app/controllers/api/v1/schedules_controller.rb

```diff
@@ -0,0 +1,34 @@
+class Api::V1::SchedulesController < ApplicationController
+
+  def show 
+    schedule = Schedule.find_by(user_id: params[:user_id])
+    if schedule 
+      render json: ScheduleSerializer.new(schedule).serializable_hash.to_json
+    else
+      render json: { error: "Schedule not found" }, status: :not_found
+    end
+  end
+
+  def update
+    schedule = Schedule.find_by(user_id: params[:user_id])
+    show = Show.find_by(id: params[:show_id])
+
+    if schedule && show
+      if params[:action_type] == 'add'
+        if schedule.shows.count < 8
```

Comment:
Try to avoid "magic" numbers and strings. Set stuff like this to a constant somewhere.